.github/workflows/pr.yml

```diff
@@ -192,22 +210,21 @@ jobs:
         uses: actions/cache@v2
         id: lein-cache
         with:
-          # note: this should coincide with ${BIN_PATH}/lein. unclear
+          # note: these should coincide with ${BIN_PATH}/lein and ${LEIN_HOME}. unclear
           # if variables can be used here
-          path: ~/bin/lein
-          key: ctia-lein-cache-${{ env.LEIN_VERSION }}
+          path: |
+            ~/bin/lein
+            ~/.lein
+          key: ctia-lein-cache-${{ env.ACTIONS_CACHE_VERSION }}-${{ env.LEIN_VERSION }}
       - name: Install Leiningen
-        # why doesn't this work!? using an if statement instead
-        #if: steps.lein-cache.cache-hit != 'true'
+        if: steps.lein-cache.outputs.cache-hit != 'true'
         run: |
-          if ! command -v lein
-          then
-            cd "${BIN_PATH}"
-            curl -sLO https://raw.githubusercontent.com/technomancy/leiningen/${LEIN_VERSION}/bin/lein
-            chmod a+x lein
-            lein version
-          fi
-
+          cd "${BIN_PATH}"
+          # poor man's travis_retry
+          for i in {1..${ACTIONS_RETRY_TIMES}}; do curl -sLO "https://raw.githubusercontent.com/technomancy/leiningen/${LEIN_VERSION}/bin/lein" && break; done
+          chmod a+x lein
+          # poor man's travis_retry
+          for i in {1..${ACTIONS_RETRY_TIMES}}; do lein version && break; done
```

Comment:
Probably you've heard of https://github.com/DeLaGuardo/setup-clojure ?

Might save some LOCs. Using it sha-versioned would constrain risks.

Reply:
Haven't had a great experience with that action in terms of caching. I can't see an indication that's improved, but IIRC caching just "didn't work".